aws-creds/iam.tf

```diff
@@ -59,7 +66,64 @@
 
 #data "aws_caller_identity" "current" {}
 
+resource "aws_iam_role" "terraform_architect_iam_role" {
+  name = "terraform-architect"
+  path = "/"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::162808325377:root"
+      },
+      "Action": "sts:AssumeRole",
+      "Condition": {}
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "attach_iam"{
+  role = aws_iam_role.terraform_architect_iam_role.name
+  policy_arn = aws_iam_policy.terraform_iam_policy.arn
+}
+
+resource "aws_iam_role_policy_attachment" "attach_iam_write"{
+  role = aws_iam_role.terraform_architect_iam_role.name
+  policy_arn = aws_iam_policy.terraform_iam_write_policy.arn
+}
 
+resource "aws_iam_group" "terraform_architects_iam_group" {
+  name = "terraform-architects"
+  path = "/"
+}
+
+resource "aws_iam_policy" "terraform_assume_iam_policy" {
+  name = "terraform-assume-deployment-role"
```

Comment:
We should have a prefix for all resources with 'name', since otherwise you can't have two deploys of this in the same AWS account.

Reply:
what kind of prefix? One we generate or more like one we put in a variable at the top of the file and use everywhere for easy duplication. I would note that this was intended to be one per account even if multiple people were using the account - one exception, IamWrite is separate from the other perms and you may want a second set to prototype with pre-made IAM roles (We certainly do)